### DISCUSSION TITLE
ztnet in allready initialized environment

### DISCUSSION THREAD

**Discussion by vargtas** [sinamics/ztnet#114] [General]
Hi.

First of all I will say thank you for a very nice app! (Tusen takk! Dette har jeg sett lenge etter :-))
I have an environment with two moons, a controller and several networks.
Is there any way for me to start using ztnet with my already initialized environment, and if so how do I import my settings to ztnet?

**Comment by sinamics** [comment]
Hi. Bra du likte app`n  😃 

## Backup
Before you test anything, backup your config as this territory has not been heavily tested.


You should be able to just point ztnet towards your zerotier controller.
Add ip and the secret for your controller. You can find the secret key in authoken.secret. `cat /var/lib/zerotier-one/authtoken.secret`
![image](https://github.com/sinamics/ztnet/assets/7771916/32e9232c-f64c-48b1-8ec1-0c53bdd10860)


Then all your networks will pop up in a table on the same page (admin=>controller). Here you need to map all network to your user account in ztnet.

**Comment by vargtas** [comment]
Done!
But the networks I already have doesn't show up under Navigation/Local Controller
I guess this has something to do with the networks not existing in the PSQL(?)
Is there any way I may populate the data to the SQL database?

**Comment by sinamics** [comment]
When you add your custom zerotier ip/port, you should see a table with all the unlinked networks that is not added to the psql database. 
Here is an example:
![image](https://github.com/sinamics/ztnet/assets/7771916/6fd9b0df-2e48-4a30-a2dc-f59638f782f8)


If you dont see any table, that means ztnet does not recognize any networks of your controller. 
After you have added your zerotier controller ip/port, do you see any number of networks ?
![image](https://github.com/sinamics/ztnet/assets/7771916/db17ab50-6ea5-4404-8928-018b988ae882)

**Reply by vargtas** [reply]
No options to add any unlinked networks despite ztnet showing 5 total networks and 20 total members.
I also tried to spin up the zerotier controller as a docker like the doc says, but still no cigar...

**Reply by sinamics** [reply]
thats odd. will test again in my dev enviroment

**Reply by sinamics** [reply]
@vargtas i guess you are using the latest zerotier version 1.12.1 ? 
Ztnet does not support that version yet due to a bug in zerotier members api, hence the table will not show up.
https://github.com/zerotier/ZeroTierOne/issues/2114

**Reply by vargtas** [reply]
You are right! Once I downgraded the zerotier version to 1.10.6 I got the option to add the networks.
Thanks a lot for the help and sorry for taking up your time by not using the correct zerotier version from the beginning.

**Reply by sinamics** [reply]
Great.I will release a fix for the zt bug, so users can experiment with latest version. It will probably be relased later today, doing all the pre-releaes testing now.
